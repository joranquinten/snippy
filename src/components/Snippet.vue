<template>
  <v-card class="snippet">
    <v-card-title>
      <h3 class="title">{{ title }}</h3>
      <span class="language">{{ language }}</span>
    </v-card-title>
    <v-card-text>
      <pre>{{ content }}</pre>
    </v-card-text>
    <v-textarea
      v-show="false"
      id="content"
      background-color="#BD93F9"
      outline
      auto-grow
      rows="1"
      name="content"
      browser-autocomplete="false"
      label="This is where the snippet goes"
      v-model="content"
      persistent-hint
      hint="Select and copy the snippet"
      @focus="$event.target.select()"
    ></v-textarea>

    <tags v-if="tags" :tags="tags"></tags>

    <v-card-actions>
      <router-link :to="`/snippet/${id}/edit`"
        ><v-btn flat>Edit</v-btn></router-link
      >
    </v-card-actions>
  </v-card>
</template>

<script>
import tags from "./Tags";

export default {
  name: "snippet",
  components: {
    tags
  },
  props: {
    id: {
      type: String,
      required: true
    },
    title: {
      type: String
    },
    content: {
      type: String
    },
    language: {
      type: String,
      default: "plaintext"
    },
    tags: {
      type: Array
    }
  }
};
</script>

<style lang="scss" scoped>
.snippet {
  .title {
    font-weight: 400;
    margin: 0;
    color: #2c3e50;

    .language {
      font-size: 0.72em;
    }
  }
}
</style>
