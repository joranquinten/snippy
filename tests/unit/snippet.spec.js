import { shallowMount } from "@vue/test-utils";
import Snippet from "@/components/Snippet.vue";

describe("Snippet.vue", () => {
  const snippets = [
    {
      id: "1",
      title: "exampleSnippet",
      content: "some snippet content",
      language: "jest",
      tags: ["some", "tags"]
    }
  ];
  it("renders the props provided to the component", () => {
    const wrapper = shallowMount(Snippet, {
      propsData: snippets[0]
    });

    expect(wrapper.find(".title").text()).toEqual(snippets[0].title);
    expect(wrapper.find("textarea").element.value).toEqual(snippets[0].content);
    expect(wrapper.find(".language").text()).toEqual(snippets[0].language);
    expect(wrapper.find(".tags")).toMatchSnapshot();
  });
});
