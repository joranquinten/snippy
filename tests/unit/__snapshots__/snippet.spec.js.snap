// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet.vue renders the props provided to the component 1`] = `
<ul class="tags">
  <li>some</li>
  <li>tags</li>
</ul>
`;
